libs/designer/src/lib/core/actions/bjsworkflow/templates.ts

```diff
@@ -193,7 +192,25 @@ const loadTemplateFromResourcePath = async (templateName: string, manifest: Temp
         data.workflows[workflowPath] = workflowData.workflow;
         data.parameterDefinitions = {
           ...data.parameterDefinitions,
-          ...workflowData.parameterDefinitions,
+          ...Object.keys(workflowData.parameterDefinitions).reduce((acc: Record<string, Template.ParameterDefinition>, key: string) => {
```

Comment:
I think this is going to run multiple times for shared parameters as well... so when you get the tests fixed, please do move this calculation after the for loop is done on the parameterDefinitions, so it only runs once for each unique parameter.